### ISSUE TITLE
Typeahead Moves to End of The Field When Typing

### ISSUE THREAD

**Issue by kkirsche** [mgcrea/angular-strap#1448]
The typeahead module doesn't respect where you are in an input and instead pulls you to the end of the input.

For example, if you type in the sample "rgi" and realize you you forgot the beginning, you can't jump to the first part of the input and add "Vi" to get "Virgi" as you type "Virginia"

Example:
https://www.youtube.com/watch?v=L7YdT8SSWm0

![typeahead](https://cloud.githubusercontent.com/assets/947110/5838195/9da21dfe-a153-11e4-8eb1-596ddad1db48.gif)
 
After further research this seems to be caused by this line in the typeahead module:
```javascript
element.val(selected ? selected.toString().replace(/<(?:.|\n)*?>/gm, '').trim() : '');
```

If we don't call `controller.$render();` in this block at the end this fixes this issue:

```javascript
// Watch model for changes
        scope.$watch(attr.ngModel, function(newValue, oldValue) {
          // console.warn('$watch', element.attr('ng-model'), newValue);
          scope.$modelValue = newValue; // Publish modelValue on scope for custom templates
          parsedOptions.valuesFn(scope, controller)
          .then(function(values) {
            // Prevent input with no future prospect if selectMode is truthy
            // @TODO test selectMode
            if(options.selectMode && !values.length && newValue.length > 0) {
              controller.$setViewValue(controller.$viewValue.substring(0, controller.$viewValue.length - 1));
              return;
            }
            if(values.length > limit) values = values.slice(0, limit);
            var isVisible = typeahead.$isVisible();
            isVisible && typeahead.update(values);
            // Do not re-queue an update if a correct value has been selected
            if(values.length === 1 && values[0].value === newValue) return;
            !isVisible && typeahead.update(values);
            // Queue a new rendering that will leverage collection loading
            controller.$render();
          });
        });
```

**Comment by bstruthers** [comment]
Interestingly enough, this doesn't happen to me with a Firefox nightly but it does in Chrome.

**Comment by kkirsche** [comment]
Interesting. Thanks for sharing that info @bstruthers

**Comment by vmlf01** [comment]
Duplicate of #1083

**Comment by meishier** [comment]
thanks. modify typeahead module, 
          var ss = element[0].selectionStart, sd = element[0].selectionEnd; //addd
          element.val(options.trimValue === false ? value : value.trim());
          element[0].setSelectionRange(ss, sd); //add
can fix it.

**Comment by mgcrea** [comment]
Fixed by #2079
